lib/idpartner.js

```diff
@@ -137,16 +137,64 @@ class IDPartner {
     };
   }
 
-  async getAuthorizationUrl(query, proofs, scope, prompt) {
+  #extractClaims(claimsObject) {
+    if (claimsObject && claimsObject.constructor === Object) {
+      let userinfo_keys = Object.keys(claimsObject.userinfo || {});
+      let id_token_keys = Object.keys(claimsObject.id_token || {});
+      return userinfo_keys.concat(id_token_keys);
+    } else {
+      return [];
+    }
+  }
+
+  async #mapClaimsToScopes(claimsObject) {
+    if (claimsObject && claimsObject.constructor === Object) {
+      // This map comes from core-services/oidc-provider-service/src/services/express/provider.js:50-66
+      // we put it here statically because we need to accept consent to obtain the iss
+      // and we need the iss to obtain the dynamic scopes and claims for the OIDC server
+      // chicken and egg problem
+      const scopesToClaims = {
+        address: ['address'],
+        email: ['email'],
+        profile: ['birthdate', 'family_name', 'given_name'],
+        age_over_18: ['age_over_18'],
+        age_over_21: ['age_over_21'],
+        age_over_25: ['age_over_25'],
+        payment_details: ['payment_details'],
+        'vc.MockBankCredential': ['vc.MockBankCredential'],
+      };
+      let claims = this.#extractClaims(claimsObject);
+
+      const claimsToScopes = Object.keys(scopesToClaims).reduce((acc, scope) => {
```

Comment:
Why do we parse from scopes to claims instead of defining a hardcoded map? Example:
```
const claimsToScopes = {
  address: 'address',
  email: 'email',
  birthdate: 'profile',
  family_name: 'profile',
  family_name: 'profile',
  etc
}
```

This would avoid complex code and would be more performant.